skeleton/src/main/java/org/killbill/commons/skeleton/modules/StatsModule.java

```diff
@@ -86,19 +86,8 @@ protected void configure() {
         final TimedResourceListener timedResourceTypeListener =
                 new TimedResourceListener(getProvider(GuiceContainer.class), getProvider(MetricRegistry.class));
         bindListener(Matchers.any(), timedResourceTypeListener);
-    }
-
-    @Provides
-    @Singleton
-    protected HealthCheckRegistry provideHealthCheckRegistry(final Set<HealthCheck> healthChecks) {
-        final HealthCheckRegistry healthCheckRegistry = createHealthCheckRegistry();
-
-        // It used to be done by AdminServletProvider. JavaDoc of AdminServletModule outdated?
-        for (final HealthCheck healthCheck : healthChecks) {
-            healthCheckRegistry.register(healthCheck.getClass().getName(), healthCheck);
-        }
 
-        return healthCheckRegistry;
+        bind(HealthCheckRegistry.class).toInstance(createHealthCheckRegistry());
```

Comment:
So if i understand correctly, we bind this `HealthCheckRegistry` to a method that return a `new KillBillHealthCheckRegistry ()` (default CTOR), at which point such object does not do anything. Later Guice finds out we annotated the method `initialize` and does the method injection, but at that time DBI was initialized ands things work.

Reply:
Yup! The registry needs to be created and passed to the DAO (HikariCP healthcheck) first. Then, when the DAO is setup, the healthchecks themselves can be initialized and injected into the registry.